### PR TITLE
Docker MySQL <= 5.7 image fix for Apple Silicone (trunk and 6.4 branches)

### PR DESCRIPTION
**Pull request by desrosj**
Trac ticket: https://core.trac.wordpress.org/ticket/59930

---
**This Pull Request is for code review only. Please keep all other discussion in the Trac ticket. Do not merge this Pull Request. See [GitHub Pull Requests for Code Review](https://make.wordpress.org/core/handbook/contribute/git/github-pull-requests-for-code-review/) in the Core Handbook for more details.**

### CODE REVIEW

File: .env [hunk 1]
```diff
@@ -46,12 +46,12 @@ LOCAL_DB_TYPE=mysql
...
 # The database version to use.
 #
-# Defaults to 5.7 with the assumption that LOCAL_DB_TYPE is set to `mysql` above.
+# Defaults to 8.0 with the assumption that LOCAL_DB_TYPE is set to `mysql` above.
 #
 # When using `mysql`, see https://hub.docker.com/r/amd64/mysql for valid versions.
 # When using `mariadb`, see https://hub.docker.com/r/amd64/mariadb for valid versions.
 ##
-LOCAL_DB_VERSION=5.7
+LOCAL_DB_VERSION=8.0
```
Comment (ockham) [inline .env:54]: This seems even more timely now that the 5.7 image [seems to have been removed from the Docker registry altogether](https://core.trac.wordpress.org/ticket/59930#comment:16):

> I just checked https://hub.docker.com/r/amd64/mysql (which is what we're referring to from the [`.env` file](https://core.trac.wordpress.org/browser/trunk/.env?rev=56453#L51), and I noticed that MySQL 5.7 isn't even listed there anymore.
>
> I dug a bit deeper and found that it was [removed recently](https://github.com/docker-library/official-images/commit/3b2ef60353adbcee746512bc0f29f71967bf5bb9); the [commit](https://github.com/docker-library/mysql/commit/3e959c2) is titled "Remove 5.7 entirely (EOL)".

File: .env [hunk 2]
```diff
@@ -46,12 +46,12 @@ LOCAL_DB_TYPE=mysql
 ##
 # The database version to use.
 #
-# Defaults to 5.7 with the assumption that LOCAL_DB_TYPE is set to `mysql` above.
+# Defaults to 8.0 with the assumption that LOCAL_DB_TYPE is set to `mysql` above.
 #
 # When using `mysql`, see https://hub.docker.com/r/amd64/mysql for valid versions.
 # When using `mariadb`, see https://hub.docker.com/r/amd64/mariadb for valid versions.
```
Comment (ockham) [inline .env:52]: Given https://github.com/WordPress/wordpress-develop/pull/5834/files#r1455897013, we might as well point to the platform-independent reference:

```suggestion
# When using `mysql`, see https://hub.docker.com/_/mysql for valid versions.
# When using `mariadb`, see https://hub.docker.com/_/mariadb for valid versions.
```

File: README.md [hunk 3]
```diff
@@ -118,6 +118,26 @@ Starting the environment again is a single command:
...
+The MySQL Docker images do not support Apple Silicone processors (M1, M2, etc.) for MySQL versions 5.7 and earlier.
+
+When using MySQL <= 5.7 on an Apple Silicone machine, the following line in the `docker-compose.yml` file will need to be uncommented:
+
+```
+platform: linux/amd64
+```
+
+Additionally, the "Use Rosetta for x86/AMD64 emulation on Apple Silicon" setting in Docker needs to be disabled for this workaround.
+
```
Comment (ockham) [inline README.md]: Might be obsolete now?

```suggestion
```

**Comment by ockham** [comment]
Hmm, weird (and somewhat worrying) that [two](https://github.com/WordPress/wordpress-develop/actions/runs/7425055173/job/20205953346?pr=5834) [of](https://github.com/WordPress/wordpress-develop/actions/runs/7425055173/job/20205951461?pr=5834) the build jobs are failing on MacOS:

image

It's "Gutenberg running from src" and "Gutenberg running from build"; the Core counterparts seem to be passing. Wondering if we also need to change some environment setting in GB? In case the build downloads some platform-specific binary dependencies maybe? 🤔

File: README.md (hunk 3 above)
Comment (desrosj) [reply README.md]: I'd like to leave this. I added this so that anyone looking to use MySQL <= 8.0 would be able to troubleshoot themselves.

**Comment by desrosj** [comment]
Merged in https://core.trac.wordpress.org/changeset/57568.

**Pull request closed without merging**